src/util/fingerprint.cpp

```diff
@@ -1,2 +1,61 @@
 #include "util/fingerprint.hpp"
-#include "util/fingerprint_impl.hpp"
+#include "util/exception.hpp"
+#include "util/exception_utils.hpp"
+#include "util/version.hpp"
+
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+
+#include <cstring>
+
+#include <algorithm>
+#include <string>
+
+namespace osrm
+{
+namespace util
+{
+
+FingerPrint FingerPrint::GetValid()
+{
+    FingerPrint fingerprint;
+
+    // 4 chars, 'O','S','R','N' - note the N instead of M, v1 of the fingerprint
+    // used M, so we add one and use N to indicate the newer fingerprint magic number.
+    // Bump this value if the fingerprint format ever changes.
+    fingerprint.magic_number = {{'O', 'S', 'R', 'N'}};
+    // fingerprint.magic_number[1] = 'S';
+    // fingerprint.magic_number[2] = 'R';
+    // fingerprint.magic_number[3] = 'N';
+    fingerprint.major_version = static_cast<std::uint8_t>(atoi(OSRM_VERSION_MAJOR));
```

Comment:
It is better to fix directly in `include/util/version.hpp.in`
```
#define OSRM_VERSION_MAJOR @OSRM_VERSION_MAJOR@
#define OSRM_VERSION_MINOR @OSRM_VERSION_MINOR@
#define OSRM_VERSION_PATCH @OSRM_VERSION_PATCH@

#define OSRM_VERSION_(A,B,C) "v" #A "." #B "." #C
#define OSRM_VERSION OSRM_VERSION_(OSRM_VERSION_MAJOR, OSRM_VERSION_MINOR, OSRM_VERSION_PATCH)
```
and here avoid str->int->char conversion